crates/services/p2p/src/p2p_service.rs

```diff
@@ -815,6 +765,12 @@ impl FuelP2PService {
                 error,
                 request_id,
             } => {
+                // If the remote peer doesn't support the protocol, it is better to disconnect
+                // to find another peer that supports it.
+                if let OutboundFailure::UnsupportedProtocols = error {
+                    let _ = self.swarm.disconnect_peer_id(peer);
```

Comment:
can we add a test that asserts the `connected_peers` table within each of the behaviours is reducing by 1 when an `UnsupportedProtocol` is detected?

Reply:
We could, but I don't know how to test it. Because after we disconnect from the peer, it is possible that we will connect to the same peer again.

So it is quite hard to set up the right condition in the test where we need to connect to a busy peer, send a request-response, and be reconnected to another peer=)